src/xdp/apikernel.c

```diff
@@ -116,42 +118,74 @@ XdpCreateProgramKernel(
     ProgramOpen.RuleCount = RuleCount;
     ProgramOpen.Rules = Rules;
 
-    return XdpProgramCreate((XDP_PROGRAM_OBJECT **)Program, &ProgramOpen, KernelMode);
+    if (!ExAcquireRundownProtectionCacheAware(Client->RundownRef)) {
```

Comment:
I don't think the XDP provider should assume an XDP "helper" struct is being passed. Typically the client and provider each maintain their own rundown objects.

Reply:
Please elaborate it?
What I learned from document is to manage InProgressCallingCount from client side only.
Any reference?